### PR TITLE
More extendable Scaffolder page 

### PR DESCRIPTION
**Pull request by alexrybch**
## Hey, I just made a Pull Request!

We recently extended scaffolder page and it would be really nice to have a possibility to change also header and subheader as well.
I also added export of routes to have a possibility to use those when creating custom `TemplateCard`

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the proposal! This PR needs an updated API report.
I will need to defer to @benjdlambert if this is a desired level of customisation for the ScaffolderPage.

**Comment by alexrybch** [comment]
> Thanks for the proposal! This PR needs an updated API report. I will need to defer to @benjdlambert if this is a desired level of customisation for the ScaffolderPage.

As an idea we can also pass scaffolder route refs inside `TemplateCard` as a prop instead of exposing them so they can be used in custom cards.

File: plugins/scaffolder/src/routes.ts [hunk 1]
```diff
@@ -19,64 +19,75 @@ import {
   createSubRouteRef,
 } from '@backstage/core-plugin-api';
 
+/** @public */
```
Comment (benjdlambert) [inline plugins/scaffolder/src/routes.ts:22]: Don't think we should be exporting the `externalRouteRefs`, they don't provide any value.

File: plugins/scaffolder/src/routes.ts [hunk 2]
```diff
@@ -19,64 +19,75 @@ import {
...
  * @deprecated This is the old template route, can be deleted before next major release
  */
+/** @public */
 export const legacySelectedTemplateRouteRef = createSubRouteRef({
   id: 'scaffolder/legacy/selected-template',
   parent: rootRouteRef,
   path: '/templates/:templateName',
 });
 
+/** @public */
```
Comment (benjdlambert) [inline plugins/scaffolder/src/routes.ts:50]: The `next*` `routeRefs` should be `@alpha` exports.

File: plugins/scaffolder/src/routes.ts [hunk 3]
```diff
@@ -19,64 +19,75 @@ import {
...
 export const rootRouteRef = createRouteRef({
   id: 'scaffolder',
 });
 
 /**
  * @deprecated This is the old template route, can be deleted before next major release
  */
+/** @public */
```
Comment (benjdlambert) [inline plugins/scaffolder/src/routes.ts:43]: This is also deprecated, so let's not export this at all.

Comment (benjdlambert) [reply plugins/scaffolder/src/routes.ts:43]: Want to remove this soon.

**Review by benjdlambert — CHANGES_REQUESTED** [review]
> As an idea we can also pass scaffolder route refs inside TemplateCard as a prop instead of exposing them so they can be used in custom cards.

I think exporting them is fine, we just need to be careful which ones we export, and how we export them. 👍

**Review by benjdlambert — COMMENTED** [review]
Think this is looking better now, just a note about the changeset

**Review by benjdlambert — APPROVED** [review]
Thanks! :pray:

File: .changeset/sixty-islands-develop.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder': patch
```
Comment (jhaals) [inline .changeset/sixty-islands-develop.md:2]:
```suggestion
'@backstage/plugin-scaffolder': minor
```
Scaffolder is version 1.x so new features should be minor changes :)

File: plugins/scaffolder/api-report.md [hunk 5]
```diff
@@ -464,6 +482,11 @@ export interface ScaffolderUseTemplateSecrets {
   setSecrets: (input: Record<string, string>) => void;
 }
 
+// @public (undocumented)
+export const selectedTemplateRouteRef: SubRouteRef<
```
Comment (jhaals) [inline plugins/scaffolder/api-report.md:486]: Is it worth mentioning these new exports in the changeset too?

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks! Just a few small fixes left

File: plugins/scaffolder/api-report.md (hunk 5 above)
Comment (alexrybch) [reply plugins/scaffolder/api-report.md:486]: Yep, missed this !

File: .changeset/sixty-islands-develop.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+---
+'@backstage/plugin-scaffolder': minor
+---
+
+<<<<<<< Updated upstream
```
Comment (benjdlambert) [inline .changeset/sixty-islands-develop.md:5]: Looks like some merge conflict here?

Comment (alexrybch) [reply .changeset/sixty-islands-develop.md]: my bad, fixed!

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**